src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -47,6 +53,43 @@ class PythonToolRequirementsBase(Subsystem):
     default_lockfile_resource: ClassVar[tuple[str, str] | None] = None
     default_lockfile_url: ClassVar[str | None] = None
 
+    install_from_resolve = StrOption(
+        advanced=True,
+        default=None,
```

Comment:
It would be great if this / rule code using it eventually got smart enough to only require saying from which resolve if you have more than one resolve. It would be really great if Pants rule code grew a step where it tried something like `pex3 lock export --lock user.lock tool-req1 tool-req2`, and if it got a success, just automatically used the user lock file.